autotelic/fastify-slonik
# Feature/slonik plugin

### Summary

- Moved [Slonik Plugin](https://github.com/autotelic/harpocrates/tree/main/plugins/fastify-slonik) from Harpocrates repo.

### Test Plan

- Clone and install `npm i`
- To confirm the plugin is working, clone and install Harpocrates project from [this repo](https://github.com/autotelic/harpocrates)
  - `cd` into the project directory and edit location of fastifySlonik in app.js
  ```js
  ...
  const fastifySlonik = require('<path to this plugin, i.e. ../fastify-slonik>')
  ...
  ```
  - Create .envrc `cp .envrc.example .envrc`
  - Update .envrc with public ip
    ```sh
    $ ifconfig -u | grep 'inet ' | grep -v 127.0.0.1 | cut -d\  -f2 | head -1
    # <your-public-ip> <-- Copy this
    ```
  - Update .envrc with copied public-ip `export PUBLIC_IP=<your-public-ip>`
  - Run the following commands to create and run [docker](https://www.docker.com/) container `docker-compose up`
  - Seed and migrate database
  ```sh
  $ cp test_seeds/1605303453216_seeds.js migrations/
  $ npm run migrate up
  ```
  - Run in development `npm run dev`
- If plugin is working correctly, GET request to `localhost:3000/users/1` should return
```js
{
userId: 1,
githubUsername: "Ed",
emailAddress: "Ed@autotelic.com",
firstLogin: 1610521099724,
mostRecentLogin: 1610521099724
}
```

### To Do
- index.test.js is failing, need to create example.js instead of requiring harpocrates app to test
